test/emailpassword/passwordreset.test.js

```diff
@@ -13,25 +13,329 @@
- *        - Call the createResetPasswordToken with unknown userId and test error thrown
- *        - password validation checks
- *        - non existent email should return "OK" with a pause > 300MS
- *        - check that the generated password reset link is correct
- *        - check if that if the getResetPasswordURL function is given by the user, that is called
- *        - check that the password link has a token and a rid
+ *        - (later) Call the createResetPasswordToken function with valid input
+ *        - (later) Call the createResetPasswordToken with unknown userId and test error thrown
+ *        - email validation checks (done)
+ *        - check that the generated password reset link is correct (done)
  * TODO: password reset API:
- *        - Call the resetPasswordUsingToken function with valid input
- *        - Call the resetPasswordUsingToken with an invalid token and see the error
- *        - token is missing from input
- *        - token is not of type string from input
- *        - invalid token in input
- *        - password is invalid in input
- *        - password is missing in input
- *        - input is valid, check that password has changed (call sign in)
+ *        - (later) Call the resetPasswordUsingToken function with valid input
+ *        - (later) Call the resetPasswordUsingToken with an invalid token and see the error
+ *        - password validation checks (done)
+ *        - token is missing from input (done)
+ *        - (later) token is not of type string from input
+ *        - invalid token in input (done)
+ *        - input is valid, check that password has changed (call sign in) (done)
  */
+
+describe(`passwordreset: ${printPath("[test/passwordreset.test.js]")}`, function () {
+    beforeEach(async function () {
+        await killAllST();
+        await setupST();
+        ProcessState.getInstance().reset();
+    });
+
+    after(async function () {
+        await killAllST();
+        await cleanST();
+    });
+
+    /*
+     *  TODO: generate token API:
+     *      - email validation checks
+     *      - non existent email should return "OK" with a pause > 300MS
+     *      - check that the generated password reset link is correct
+     */
+    it("test email validation checks in generate token API", async function () {
+        await startST();
+        STExpress.init({
+            supertokens: {
+                connectionURI: "http://localhost:8080",
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [EmailPassword.init()],
+        });
+        let emailpassword = await EmailPasswordRecipe.getInstanceOrThrowError();
+        const app = express();
+
+        app.use(STExpress.middleware());
+
+        app.post("/reset-password-invalid-email", async (req, res) => {
```

Comment:
# Major issue
- You do not need to create your own API (unless the tests requires that). The middleware above adds it on its own. You simply need to call the default API below. 
- The request body must be given in the request.. in set in the API.
- Requests do not have a validation function in the formFields.. Please see the types properly..

I think you will need to redo all tests that are written this way.